sql/src/main/java/io/crate/operation/collect/sources/InformationSchemaCollectSource.java

```diff
@@ -156,9 +161,12 @@ public void remove() {
     @SuppressWarnings("unchecked")
     public Collection<CrateCollector> getCollectors(CollectPhase collectPhase, RowReceiver downstream, JobCollectContext jobCollectContext) {
         Routing routing =  collectPhase.routing();
-        assert routing.locations().containsKey(TableInfo.NULL_NODE_ID);
-        assert routing.locations().get(TableInfo.NULL_NODE_ID).size() == 1;
-        String fqTableName = routing.locations().get(TableInfo.NULL_NODE_ID).keySet().iterator().next();
+        Map<String, Map<String, List<Integer>>> locations = routing.locations();
+        assert locations != null;
+        String localNodeId = clusterService.localNode().id();
```

Comment:
use ```clusterService.state().nodes().localNodeId()``` the code above might lead to errors, when the node is leaving the cluster...